### ISSUE TITLE
kubernetes filter unescapes log message

### ISSUE THREAD

**Issue by sh0rez** [fluent/fluent-bit#965]
## Bug Report

**Describe the bug**
When feeding logfmt formatted messages which contain escaped hyphens into the kubernetes filter, the escaping gets removed which results in the logfmt parser breaking.

My application logs json in as a field in logfmt and escapes it properly. Fluentbit takes the logs from docker which removes the escapings added by docker but leaves mine intact. As soon as it hits the kubernetes filter, my escapings are gone. The logfmt then creates a lot of junk fields as the hyphens do not match anymore.

This only happens if enabling `Merge_Log` on the filter which seems to be a requirement for `K8S-Logging.Parser` parsers to work?. I tried disabling `Merge_Log` which indeed fixes the escaping, but the fields are not parsed anymore (or at least they don't make their way to the elasticsearch)

**To Reproduce**
- Example log message if applicable:
```
{"log":"time=\"2018-12-16T10:20:59Z\" level=warning msg=\"metric value parse failure\" domain=abschaltungen field=16-12-18-07-39-13 value=\"{\\\"value\\\":\\\"keine Anf.\\\"}\"\n","stream":"stderr","time":"2018-12-16T10:20:59.868231694Z"}
```
- Steps to reproduce the problem:
1. Run some application logging in logfmt and putting escaped json into a field (e.g. https://github.com/sh0rez/luxtronik2-exporter)
2. Run it on kubernetes, configure FluentBit to process all logs from the docker daemon, hand them to the kubernetes filter and specify the logfmt parser using annotation.

**Expected behavior**
The escaping is left intact by the kubernetes filter, the logfmt parser creates a field called `value` with the value `{"value":"keine Anf."}`

**Screenshots**
![kibana_fluentbug](https://user-images.githubusercontent.com/26971117/50052787-e8a62400-0129-11e9-9916-569beefa8686.png)
One can see that there are weird junk fields, e.g. `:`, `284h`, `}` and especially `value` just got `true` instead of the real value.

**Your Environment**

* Version used:
docker image `fluent/fluent-bit:1.0.0`
* Configuration:
<details><summary>fluent-bit.conf</summary>

```ini
[SERVICE]
    Flush         1
    Log_Level     info
    Daemon        off
    Parsers_File  parsers.conf
    HTTP_Server   On
    HTTP_Listen   0.0.0.0
    HTTP_Port     2020

[INPUT]
    Name              tail
    Tag               kube.*
    Path              /var/log/containers/*.log
    Parser            docker
    DB                /var/log/flb_kube.db
    Mem_Buf_Limit     5MB
    Skip_Long_Lines   On
    Refresh_Interval  10

[FILTER]
    Name                kubernetes
    Match               kube.*
    Merge_Log           On
    K8S-Logging.Parser  On
    K8S-Logging.Exclude On

[OUTPUT]
    Name            es
    Match           *

    Host            xxx
    Port            xxx
    tls  On

    Index           xxx
    Logstash_Format Off

    Retry_Limit     False
```

</details>
<details><summary>parsers.conf</summary>

```ini
# logfmt used by logrus and other go applications
[PARSER]
    Name   logfmt
    Format logfmt

[PARSER]
    Name        docker
    Format      json
    Time_Key    time
    Time_Format %Y-%m-%dT%H:%M:%S.%L
    Time_Keep   On
    # Command      |  Decoder | Field | Optional Action
    # =============|==================|=================
    Decode_Field_As   escaped    log
```

</details>

* Environment name and version (e.g. Kubernetes? What version?):
Running on Minikube:
```bash
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.3", GitCommit:"435f92c719f279a3a67808c80521ea17d5715c66", GitTreeState:"clean", BuildDate:"2018-11-26T12:57:14Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-26T16:44:10Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

* Operating System and version:
`Linux minikube 4.15.0 #1 SMP Sat Dec 8 00:26:02 UTC 2018 x86_64 GNU/Linux`

**Additional context**
I am trying to get dynamic structured logging. I want to run fluentbit as a daemonset on kubernetes and specify the parser used for the individual pods using their annotations. This works so far, but I recognized junk fields created by the logfmt parser. This is related to the kubernetes filter breaking the escaping of the logfmt fields resulting in the logfmt parser parsing rubbish.

**Comment by epcim** [comment]
@edsiper what's your ETA to for this to be fixed? We challenge this as well, I guess this hits many people as many apps do json as output.

To what is written above, Merge_Log is not a requirement if I do understand, by default is Off however its required if `log field content is a JSON string map` and you want to add these JSON key/values to the log entry.

**Comment by kiich** [comment]
I've also run into this issue and have tried:

- `Merge_Log` set to `Off`
- `Docker_Mode` set to `On`

Problem of my java app's stack trace show these still remain:
```
 	 kubernetes_cluster:	 mine
  	 log:	 u0009at org.springframework.boot.loader.Launcher.launch(Launcher.java:87)

  	 stream:	 stdout	
```
where `u0009` is not escaped.

Also multiline java stack traces are being indexed as individual events rather than all put together nicely wrapped up as 1 single event (in splunk).

**Comment by edsiper** [comment]
Please check the following comment on #1278 : 

https://github.com/fluent/fluent-bit/issues/1278#issuecomment-499583503

**Comment by edsiper** [comment]
Issue already fixed, ref: #1278 (comment)
